packages/cli/README-source.md

```diff
@@ -1222,17 +1223,22 @@ various kinds of errors occur.
 ### General Errors
 
 Errors due to a misconfiguration in a user's Zap should be handled in your app
-by throwing a standard JavaScript `Error` with a user-friendly message.
-Typically, this will be prettifying 4xx responses or APIs that return errors as
-200s with a payload that describes the error.
+by throwing `z.errors.Error` with a user-friendly message and optional error and
+status code. Typically, this will be prettifying 4xx responses or APIs that return
+errors as 200s with a payload that describes the error.
 
-Example: `throw new Error('Your error message.');`
+Example: `throw new z.errors.Error('Contact name is too long.', 'InvalidData', 400);`
 
 A couple best practices to keep in mind:
 
-  * Elaborate on terse messages. "not_authenticated" -> "Your API Key is invalid. Please reconnect your account."
-  * If the error calls out a specific field, surface that information to the user. "Invalid Request" -> "contact name is invalid"
-  * If the error provides details about why a field is invalid, add that in too! "contact name is invalid" -> "contact name is too long"
+  * Elaborate on terse messages. "InvalidData" -> "Provided data is invalid"
```

Comment:
I'd re-add the `"not_authenticated" -> "Your API Key is invalid. Please reconnect your account."` line somewhere, which helps illustrate turning a code-like thing into actual directions for the user. `InvalidData` already communicates that `data is invalid`

Reply:
The reason I changed the first one is so hat it tied back to the code argument of the example, but it doesn't add much so I'm happy to change it back.